### PR TITLE
CORE-15129: Added solution to apple tutorial

### PR DESCRIPTION
**Pull request by christian-napoles-r3**
https://r3-cev.atlassian.net/browse/CORE-15129

Testing done locally, output of tests attached to ticket linked above

Approach is to use the code solution as prescribed in the tutorial. Any additions/changes should be done in a separate ticket

### CODE REVIEW

File: contracts/build.gradle [hunk 1]
```diff
@@ -75,7 +75,7 @@ cordapp {
     // The cordapp section contains either a workflow or contract subsection depending on the type of component.
     // Declares the type and metadata of the CPK (this CPB has one CPK).
     contract {
```
Comment (christian-napoles-r3) [inline contracts/build.gradle]: question: can we include multiple workflows or only ony workflow

File: contracts/build.gradle [hunk 2]
```diff
@@ -75,7 +75,7 @@ cordapp {
...
-        name "ContractsModuleNameHere"
```
Comment (tlawson3) [inline contracts/build.gradle]: I think this should stay as it was

File: workflows/build.gradle [hunk 3]
```diff
@@ -74,7 +74,7 @@ cordapp {
     // The cordapp section contains either a workflow or contract subsection depending on the type of component.
     // Declares the type and metadata of the CPK (this CPB has one CPK).
     workflow {
-        name "WorkflowsModuleNameHere"
+        name "Apples utxo example workflow"
```
Comment (tlawson3) [inline workflows/build.gradle]: I think this should stay as it was

File: contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package com.r3.developers.apples.contracts
+
+import com.r3.developers.apples.states.AppleStamp
+import net.corda.v5.ledger.utxo.Contract
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
+
+class AppleStampContract: Contract {
+    override fun verify(transaction: UtxoLedgerTransaction) {
+        // Extract the command from the transaction
+        // Verify the transaction according to the intention of the transaction
```
Comment (tlawson3) [inline contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt:10]: I'd like to see some contract tests covering this

**Review by tlawson3 — CHANGES_REQUESTED** [review]
Do you know when you'll have the Java equivalent?
I'd like to keep these in sync as much as possible

File: contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                val output = transaction.getOutputStates(AppleStamp::class.java).first()
+                require(transaction.outputContractStates.size == 1) {
+                    "This transaction should only have one AppleStamp state as output"
+                }
+                require(output.stampDesc.isNotBlank()) {
+                    "The output AppleStamp state should have clear description of the type of redeemable goods"
+                }
+            }
+            is AppleCommands.Redeem -> {
+                // Transaction verification will happen in BasketOfApplesContract
```
Comment (tlawson3) [inline contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt:22]: This is interesting.
Is there a way I could redeem an apples stamp with an invalid transaction?
Right now we can only redeem a stamp for a basket of apples, sure; but what if in future Apple Farm decides to sell apples by the bag, and individually, as well as by the basket? Just hypothetical. Maybe this will be answered by Matt

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction,
+            "Incorrect type of BasketOfApples commands: com.r3.developers.apples.contracts." +
+                    "BasketOfApplesContractPackBasketCommandTest\$unknownCommand\$MyDummyCommand"
+        )
+    }
+
+
+
```
Comment (yarivus) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt]: remove redundant newlines

**Review by yarivus — COMMENTED** [review]
I see some pending comments

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/AppleStampContractIssueCommandTest.kt [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    fun outputContractStateSizeNotOne(){
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.Issue())
+            addOutputState(outputAppleStampState)
+            addOutputState(outputAppleStampState)
+            addSignatories(outputAppleStampStateParticipants)
+        }
+        assertFailsWith(
+            transaction,
+            "This transaction should only have one AppleStamp state as output"
```
Comment (tlawson3) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/AppleStampContractIssueCommandTest.kt:30]: Not a must, but you might find this relevant: https://github.com/corda/CSDE-cordapp-template-kotlin/blob/dcb0a2b77a170065558820cc38468f57f551225f/contracts/src/main/kotlin/com/r3/developers/csdetemplate/utxoexample/contracts/ChatContract.kt#L11-L28

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    fun unknownCommand(){
+        class MyDummyCommand: Command
+        val transaction = buildTransaction {
+            addCommand(MyDummyCommand())
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction,
+            "Incorrect type of BasketOfApples commands: com.r3.developers.apples.contracts." +
+                    "BasketOfApplesContractPackBasketCommandTest\$unknownCommand\$MyDummyCommand"
```
Comment (tlawson3) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt:106]: Maybe the contract testing lib needs to have some kind of `assertFailsWithMessageContaining` to allow you to only specify part of the message.

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/AppleStampContractIssueCommandTest.kt (hunk 7 above)
Comment (christian-napoles-r3) [reply contracts/src/test/kotlin/com/r3/developers/apples/contracts/AppleStampContractIssueCommandTest.kt:30]: I agree that this should be the standard.

I wanted to apply this practice in a future PR and I chose to stick with the current way to match what is written in the guide.

The idea is to sync a PR with the document changes to the tutorial so that no one gets mixed up

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt (hunk 8 above)
Comment (christian-napoles-r3) [reply contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt:106]: Oooh that'd a good feature!! 😁 I'll have a think about what people would want in the background

File: contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package com.r3.developers.apples.states
+
+import com.r3.developers.apples.contracts.AppleStampContract
+import net.corda.v5.ledger.utxo.BelongsToContract
+import net.corda.v5.ledger.utxo.ContractState
+import java.security.PublicKey
+import java.util.*
+
+@BelongsToContract(AppleStampContract::class)
+class AppleStamp(
```
Comment (yarivus) [inline contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt:10]:
```suggestion
class AppleStampState(
```

File: contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    val id: UUID,
+    val stampDesc: String,
+    val issuer: PublicKey,
+    val holder: PublicKey,
+    private val participants: List<PublicKey>
+): ContractState {
+    override fun getParticipants(): List<PublicKey> = participants
+
```
Comment (yarivus) [inline contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt:18]:
```suggestion
    override fun getParticipants(): List<PublicKey> = participants
```

File: contracts/src/main/kotlin/com/r3/developers/apples/states/BasketOfApples.kt [hunk 11]
```diff
@@ -0,0 +1,22 @@
+package com.r3.developers.apples.states
+
+import com.r3.developers.apples.contracts.BasketOfApplesContract
+import net.corda.v5.ledger.utxo.BelongsToContract
+import net.corda.v5.ledger.utxo.ContractState
+import java.security.PublicKey
+
+@BelongsToContract(BasketOfApplesContract::class)
+class BasketOfApples(
```
Comment (yarivus) [inline contracts/src/main/kotlin/com/r3/developers/apples/states/BasketOfApples.kt:9]:
```suggestion
class BasketOfApplesState(
```

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractRedeemCommandTest.kt [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    internal val outputBasketOfApplesState: BasketOfApples = BasketOfApples(
+        outputBasketOfApplesStateDescription,
+        outputBasketOfApplesStateFarm,
+        outputBasketOfApplesStateOwner,
+        outputBasketOfApplesStateWeight,
+        outputBasketOfApplesStateParticipants
+    )
+
+
+    @Suppress("UNCHECKED_CAST")
```
Comment (yarivus) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractRedeemCommandTest.kt]:
```suggestion
    )

    @Suppress("UNCHECKED_CAST")
```

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractRedeemCommandTest.kt [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+    @Test
+    fun happyPath(){
+        val existingAppleStampState = createInputStateAppleStamp()
+        val existingBasketOfApplesState = createInputStateBasketOfApples()
+        val transaction = buildTransaction {
+            addInputStates(listOf(existingAppleStampState.ref,existingBasketOfApplesState.ref))
+            addCommand(AppleCommands.Redeem())
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
```
Comment (yarivus) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractRedeemCommandTest.kt]:
```suggestion
        val transaction = buildTransaction {
            addInputStates(listOf(existingAppleStampState.ref,existingBasketOfApplesState.ref))
            addCommand(AppleCommands.Redeem())
            addOutputState(outputBasketOfApplesState)
            addSignatories(outputBasketOfApplesStateParticipants)
        }
```

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    @Suspendable
+    override fun call(requestBody: ClientRequestBody): String {
+        val request = requestBody.getRequestBodyAs(jsonMarshallingService, PackApplesRequest::class.java)
+        val appleDescription = request.appleDescription
+        val weight = request.weight
+        val notary = notaryLookup.notaryServices.single()
+        val myKey = memberLookup.myInfo().ledgerKeys.first()
+
+        // Building the output BasketOfApples state
+        val basket = BasketOfApples(
```
Comment (yarivus) [inline workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt]: Why not rename `BasketOfApples` => `ApplesBasket` ?

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/RedeemApplesResponderFlow.kt [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+        utxoLedgerService.receiveFinality(session) { transaction ->
+            /*
+             * [receiveFinality] will automatically verify the transaction and its signatures before signing it.
+             * However, just because a transaction is contractually valid doesn't mean we necessarily want to sign.
+             * What if we don't want to deal with the counterparty in question, or the value is too high,
+             * or we're not happy with the transaction's structure? [UtxoTransactionValidator] (the lambda created
+             * here) allows us to define the additional checks. If any of these conditions are not met,
+             * we will not sign the transaction - even if the transaction and its signatures are contractually valid.
+             */
+        }
```
Comment (yarivus) [inline workflows/src/main/kotlin/com/r3/developers/apples/workflows/RedeemApplesResponderFlow.kt:28]: Only comments in the block ... shouldn't we have code ?

**Review by yarivus — CHANGES_REQUESTED** [review]

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt (hunk 8 above)
Comment (christian-napoles-r3) [reply contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt:106]: https://r3-cev.atlassian.net/browse/CORE-15193

File: contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt (hunk 9 above)
Comment (christian-napoles-r3) [reply contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt:10]: This is an interesting discussion.

I asked in the masterclass today with @MrMatthewLayton about the naming convention for States. Whilst it is not set in stone, one school of thought is not to explicitly say State for the states.

For now:
- like most of my responses to the other comments, I will follow the same practice/standard the Apples tutorial gives me so that people who follow the guides can see line-by-line the exact same thing expected of them.
- in this case: [the code in the guide](https://docs.r3.com/en/platform/corda/5.0/developing-applications/basic-cordapp/state.html#add-imports) is what I'll follow

For the future:
- I will make another PR to update the apples tutorial solution to be written in the best practices we decide to go forward with. Yes, we should agree together what we wish to _encourage_ developers to use

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt (hunk 14 above)
Comment (christian-napoles-r3) [reply workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt]: For now, I will follow what is described in [Building your first Cordapp > Write States](https://docs.r3.com/en/platform/corda/5.0/developing-applications/basic-cordapp/state.html#check-your-work) so that it matches the tutorial, 

I am happy to change it if we agree so in the future PR when I update the solution to an improved standard

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/RedeemApplesResponderFlow.kt (hunk 15 above)
Comment (christian-napoles-r3) [reply workflows/src/main/kotlin/com/r3/developers/apples/workflows/RedeemApplesResponderFlow.kt:28]: I'm following the prescribed steps in [Building Your First CordApp > Write Flows > Check Your Work](https://docs.r3.com/en/platform/corda/5.0/developing-applications/basic-cordapp/flows.html#finalize-the-transaction-in-createandissueapplestampresponderflow) and it suggests to leave the flow blank for now.

Unless you think we should write the flow?

**Review by yarivus — COMMENTED** [review]
Checkpoint review ...
@christian-napoles-r3 please avoid squashing as it's very hard to follow incremental changes in the code review

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt (hunk 14 above)
Comment (yarivus) [reply workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt]: If we change both documentation (totally possible) and the samples code I think it's worthy ... 
Please follow-up with a new JIRA and update the docs requirements tab of that new JIRA issue such that @colin-hillis and his team are aware

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt
Comment (christian-napoles-r3) [inline workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt:1]: `PackageApples.kt` => `PackageApplesFlow.kt`

File: workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt (hunk 14 above)
Comment (christian-napoles-r3) [reply workflows/src/main/kotlin/com/r3/developers/apples/workflows/PackageApples.kt]: https://r3-cev.atlassian.net/browse/CORE-15314 - kotlin
https://r3-cev.atlassian.net/browse/CORE-15315 - java

Will do. I'll be in touch with colin's team too!

File: contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt (hunk 5 above)
Comment (christian-napoles-r3) [reply contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt:22]: documented in https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4538564702/New+Joiner+DevEx+Developers+-+Feedback so that this thought doesn't get lost. resolving conversation

File: contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt (hunk 9 above)
Comment (christian-napoles-r3) [reply contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt:10]: captured in https://r3-cev.atlassian.net/browse/CORE-15314 and https://r3-cev.atlassian.net/browse/CORE-15315

one to do in a future PR.

resolving

**Comment by christian-napoles-r3** [comment]
I'll be sure not to squash for incremental changes during code reviews.

Apologies for not having set the changes as in draft whilst I was updating them. I'll be sure not to do this again in the future :)!

File: contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+import java.security.PublicKey
+import java.util.*
+
+@BelongsToContract(AppleStampContract::class)
+class AppleStamp(
+    val id: UUID,
+    val stampDesc: String,
+    val issuer: PublicKey,
+    val holder: PublicKey,
+    private val participants: List<PublicKey>
```
Comment (MrMatthewLayton) [inline contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt:15]: You should never pass `participants` as a constructor argument (it's bad practice because you can then just pass in whoever you want). You should pass in well defined public keys (i.e. issuer, owner, debtor, creditor, etc) for _possible_ participants, and then let `getParticipants` compute which keys should be participants.

File: contracts/src/main/kotlin/com/r3/developers/apples/states/BasketOfApples.kt [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    val description: String,
+    val farm: PublicKey,
+    val owner: PublicKey,
+    val weight: Int,
+    private val participants: List<PublicKey>
```
Comment (MrMatthewLayton) [inline contracts/src/main/kotlin/com/r3/developers/apples/states/BasketOfApples.kt:14]: As per previous comment...

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/ApplesContractTest.kt [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+    )
+
+    @Suppress("UNCHECKED_CAST")
+    protected fun createInputStateAppleStamp(outputState: AppleStamp = outputAppleStampState): StateAndRef<AppleStamp> {
+        val transaction = buildTransaction {
+            addOutputState(outputState)
+            addCommand(AppleCommands.Issue())
+            addSignatories(outputState.participants)
+        }
+        transaction.toLedgerTransaction()
```
Comment (MrMatthewLayton) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/ApplesContractTest.kt]: What is this line doing? You're not using the resulting `UtxoLedgerTransaction` anywhere?

You could use it however, and get rid of those `@Suppress("UNCHECKED_CAST")` suppressions:

```kotlin
return transaction.toLedgerTransaction().getOutputStateAndRefs(AppleStamp::class.java).single()
```

Note the use of `single` here instead of `first`. If you are expecting that your transaction will only contain a single element, it's better to use `single` (and throw an exception when that's not the case), instead of using `first`.

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/ApplesContractTest.kt [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Suppress("UNCHECKED_CAST")
+    protected fun createInputStateBasketOfApples(outputState: BasketOfApples = outputBasketOfApplesState): StateAndRef<BasketOfApples> {
+        val transaction = buildTransaction {
+            addOutputState(outputState)
+            addCommand(AppleCommands.PackBasket())
+            addSignatories(outputState.participants)
+        }
+        transaction.toLedgerTransaction()
+        return transaction.outputStateAndRefs.first() as StateAndRef<BasketOfApples>
```
Comment (MrMatthewLayton) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/ApplesContractTest.kt]: As above.

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+        assertFailsWith(transaction, "The output BasketOfApples state should have non zero weight")
+    }
+
+    @Test
+    fun missingCommand() {
+        val transaction = buildTransaction {
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "List is empty.")
```
Comment (MrMatthewLayton) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt:91]: This isn't a particularly useful response when a command is missing.

**Review by MrMatthewLayton — COMMENTED** [review]
Might be worth refactoring the contract to align with the knowledge transfer that came from the first instalment of the CorDapp masterclass calls.

**Review by yarivus — COMMENTED** [review]
pending comments

File: contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt (hunk 16 above)
Comment (christian-napoles-r3) [reply contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt:15]: The best practice explanation makes sense, thanks!

Captured and will be done in https://r3-cev.atlassian.net/browse/CORE-15314 and https://r3-cev.atlassian.net/browse/CORE-15315

For now:
- like most of my responses to the other comments, I will follow the same practice/standard the Apples tutorial gives me so that people who follow the guides can see line-by-line the exact same thing expected of them.
- in this case: [the code in the guide](https://docs.r3.com/en/platform/corda/5.0/developing-applications/basic-cordapp/state.html#add-the-required-properties) is what I'll follow

For the future:
- I will make another PR to update the apples tutorial solution to be written in the best practices we decide to go forward with.

File: contracts/src/main/kotlin/com/r3/developers/apples/states/BasketOfApples.kt (hunk 17 above)
Comment (christian-napoles-r3) [reply contracts/src/main/kotlin/com/r3/developers/apples/states/BasketOfApples.kt:14]: Captured and will be done in https://r3-cev.atlassian.net/browse/CORE-15314 and https://r3-cev.atlassian.net/browse/CORE-15315

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt (hunk 20 above)
Comment (christian-napoles-r3) [reply contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt:91]: to be resolved in https://r3-cev.atlassian.net/browse/CORE-15212 :)

Comment (christian-napoles-r3) [reply contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt:91]: there was no handling in contracts tutorial for missing inputs. it will be improved 👍!

**Comment by christian-napoles-r3** [comment]
Thanks for understanding that the comments are resolved for improvements. The scope is to follow the tutorial to the T.

Improvements will definetely be done and soon with the following tickets 😁!

**Review by yarivus — APPROVED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM

File: contracts/src/test/kotlin/com/r3/developers/apples/contracts/AppleStampContractIssueCommandTest.kt [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+    @Test
+    fun unknownCommand() {
+        class MyDummyCommand : Command;
+
+        val transaction = buildTransaction {
+            addCommand(MyDummyCommand())
+            addOutputState(outputAppleStampState)
+            addSignatories(outputAppleStampStateParticipants)
+        }
+        assertFailsWith(
```
Comment (tlawson3) [inline contracts/src/test/kotlin/com/r3/developers/apples/contracts/AppleStampContractIssueCommandTest.kt:73]: Want to use the new `assertFailsWithMessageContaining` ?

**Review by tlawson3 — APPROVED** [review]
I know you're going to move replace the assertion messages with the strings from the contract themselves in another PR, so my other suggestion can also wait if you want.
